lein-finagle-clojure/src/leiningen/finagle_clojure.clj

```diff
@@ -65,14 +65,28 @@
 (defn- external-includes
   "Returns the paths of included thrift files which are not in the collection
   of source thrift file paths given."
-  [root source-paths]
+  [root source-paths thrift-dependencies]
   (let [include-paths (->> source-paths
                            (mapcat (comp scrape-includes slurp))
                            (set))
         relative-sources (->> source-paths
                               (map #(subs % (inc (count (str root)))))
-                              (set))]
-    (set/difference include-paths relative-sources)))
+                              (set))
+        external-deps (set/difference include-paths relative-sources)]
+    ;; For each of those external dependencies, recursively scrape
+    ;; THEM to find any transitive dependencies. Breadth-first search.
+    (loop [q (into clojure.lang.PersistentQueue/EMPTY external-deps)
+           result []]
```

Comment:
Seems like `result` should be a set and you should not re-enqueue scraped includes which are already part of the set. Actually, you could initialize `result` to `external-deps`.